pkg/descheduler/descheduler.go

```diff
@@ -116,8 +116,9 @@ func (d *descheduler) runDeschedulerLoop(ctx context.Context, nodes []*v1.Node)
 		metrics.DeschedulerLoopDuration.With(map[string]string{}).Observe(time.Since(loopStartDuration).Seconds())
 	}(time.Now())
 
+	shouldCheckMinNodes := !ptr.Deref(d.deschedulerPolicy.IgnoreMinNodesCheck, false)
 	// if len is still <= 1 error out
-	if len(nodes) <= 1 {
+	if shouldCheckMinNodes && len(nodes) <= 1 {
```

Comment:
I wonder whether we could limit this check only for plugins with Balance extension point. Normally, any Descheduler EP should be allowed to operator over any node independent of how many there are.

From https://github.com/kubernetes-sigs/descheduler/issues/1347:
```
It is useful to me to automatically delete pods with too many restart or failed pods.
```

Reply:
Good feedback. That makes sense to me

I can take a stab at this soon

Reply:
Thx. Lemme know once you need a review. I'll take a look.